examples/ultrasonic_sensor_hcsr04.rs

```diff
@@ -0,0 +1,51 @@
+use esp_idf_hal::delay::FreeRtos;
+use esp_idf_hal::gpio::*;
+use esp_idf_hal::peripherals::Peripherals;
+use log::*;
+use esp_idf_svc::systime::EspSystemTime;
+
+fn main() {
+    // Initialize logging and necessary peripherals
+    esp_idf_sys::link_patches();
+    esp_idf_svc::log::EspLogger::initialize_default();
+    info!("Hello, world!");
+
+    let peripherals = Peripherals::take().unwrap();
+
+    // Configure pins for trigger and echo
+    let mut trigger_pin = PinDriver::output(peripherals.pins.gpio4).expect("Error configuring trigger pin");
+    let echo_pin = PinDriver::input(peripherals.pins.gpio5).expect("Error configuring echo pin");
+
+
+    loop {
+        // Send a 10us pulse to the trigger pin to start the measurement
+        trigger_pin.set_high().expect("Error: Unable to set trigger pin high");
+        FreeRtos::delay_us(10);
+        trigger_pin.set_low().expect("Error: Unable to set trigger pin low");
+
+        // Wait for the echo pin to go high (start of pulse)
+        while !echo_pin.is_high() {}
+
+        // Measure the duration of the echo pulse (in microseconds)
+        let start_time = EspSystemTime {}.now().as_micros();
+        while echo_pin.is_high() {}
```

Comment:
You are busy looping. Isn't this triggering the TWDT? An even more interesting example might be an async one, as it would avoid the busy looping, and the waits there are expressed trivially.